### PR TITLE
Detect prefilter setupu8 4335 v7

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4112
https://redmine.openinfosecfoundation.org/issues/4335

Describe changes:
- Adds error for impossible rules such as >255 for uint8 fields
- converts icode to use the generic (many more keywords to do)
  * allows <> syntax for uint ranges to support existing syntax
  * corrects reverted ranges and issues a warning (so 20<>8 becomes 8<>20 ) to support existing syntax

Modifies #5973 by rebasing and improving error message

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
| field | test | baseline | % |
|--|--|--|--|
| | tlpr1_stats_chk |
| .app_layer.flow.ftp-data | 161 | 489 | 33.0% |

**Comment by suricata-qa** [comment]
WARNING:
| field | test | baseline | % |
|--|--|--|--|
| | tlpr1_stats_chk |
| .flow.spare | 6767226 | 5651091 | 84.0% |
| .flow.memuse | 1269129728 | 1065602688 | 84.0% |
| .tcp.insert_list_fail | 255 | 219 | 86.0% |
| .app_layer.flow.ftp-data | 161 | 489 | 33.0% |

**Comment by suricata-qa** [comment]
WARNING:
| field | test | baseline | % |
|--|--|--|--|
| | tlpr1_stats_chk |
| .flow.spare | 6767226 | 5651091 | 84.0% |
| .flow.memuse | 1269129728 | 1065602688 | 84.0% |
| .tcp.insert_list_fail | 255 | 219 | 86.0% |
| .app_layer.flow.ftp-data | 161 | 489 | 33.0% |

**Comment by catenacyber** [comment]
> ERROR: ASAN TEST FAIL in tlpr1_asan_suri QA test
> 
> Pipeline 4586

I get a 404 for https://gitlab.oisf.net/dev/suricata/-/pipelines/4586
@ct0br0 did I miss something ?

File: src/detect-engine-uint.c [hunk 1]
```diff
@@ -211,9 +211,11 @@ DetectU32Data *DetectU32Parse (const char *u32str)
...
 
                 SCLogDebug("u32 is %"PRIu32" to %"PRIu32"", u32da.arg1, u32da.arg2);
-                if (u32da.arg1 >= u32da.arg2) {
-                    SCLogError(SC_ERR_INVALID_SIGNATURE, "Invalid u32 range. ");
-                    return NULL;
+                if (u32da.arg1 > u32da.arg2) {
+                    uint32_t temp = u32da.arg1;
+                    u32da.arg1 = u32da.arg2;
+                    u32da.arg2 = temp;
+                    SCLogWarning(SC_WARN_POOR_RULE, "Reversed u32 range. ");
```
Comment (victorjulien) [inline src/detect-engine-uint.c:218]: I think this is a clear error that we shouldn't accept and fix up, we should just invalidate the rule

**Comment by victorjulien** [comment]
I'm taking the first couple of commits in the next `next` branch. The icode conflicts will need to be resolved in a future PR, as well as the poor rule warning (see my comment)

**Comment by victorjulien** [comment]
Partially merged in #6642. The rest needs a bit of work.

File: src/detect-engine-uint.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine-uint.c:218]: This was forced by unit test DetectICodeMatchTest01 :
```
    s = DetectEngineAppendSig(de_ctx, "alert icmp any any -> any any (icode:8<>20; sid:4;)");
    FAIL_IF_NULL(s);

    s = DetectEngineAppendSig(de_ctx, "alert icmp any any -> any any (icode:20<>8; sid:5;)");
    FAIL_IF_NULL(s);
```
Should we change this unit test then ?

Comment (victorjulien) [reply src/detect-engine-uint.c:218]: yeah lets not make the code bad because we have bad tests :)

Comment (catenacyber) [reply src/detect-engine-uint.c:218]: Ok will do.
I did not assume the test was bad ;-)

**Pull request closed without merging**